driver/driver_test.go

```diff
@@ -123,6 +131,41 @@ var _ = Describe("Contrail Network Driver", func() {
 		Expect(err).ToNot(HaveOccurred())
 	})
 
+	Specify("creating vswitch for forwarding extension works", func() {
+		err := contrailDriver.StartServing()
+		Expect(err).ToNot(HaveOccurred())
+
+		network, err := hns.GetHNSNetworkByName(common.RootNetworkName)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(network).ToNot(BeNil())
+
+		Expect(network.Name).To(Equal(common.RootNetworkName))
+
+		By("root network is not deleted upon shutdown of driver")
+		err = contrailDriver.StopServing()
+		Expect(err).ToNot(HaveOccurred())
+
+		_, err = hns.GetHNSNetworkByName(common.RootNetworkName)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("if root network exists upon driver startup, additional one is not created")
+		netsBefore, err := hns.ListHNSNetworks()
+		Expect(err).ToNot(HaveOccurred())
+
+		err = contrailDriver.StartServing()
+		Expect(err).ToNot(HaveOccurred())
+		_, err = hns.GetHNSNetworkByName(common.RootNetworkName)
+		Expect(err).ToNot(HaveOccurred())
+
+		netsAfter, err := hns.ListHNSNetworks()
+		Expect(err).ToNot(HaveOccurred())
+
```

Comment:
Regarding your second question - Golang has no default parameters. Weird I know

The reason for comparing lengths of lists of networks is to make sure that a duplicate wasn't created.